File: crates/klyron_plugin/bindings/cpp/builder.cpp
#include "builder.hpp"

namespace klyron::_plugin {

PluginBuilder::PluginBuilder() : config_{} {}

PluginBuilder& PluginBuilder::set_config(const PluginConfig& cfg) {
    config_ = cfg;
    return *this;
}

PluginConfig PluginBuilder::config() const {
    return config_;
}

}

File: crates/klyron_plugin/bindings/rust/builder.rs
//! Builder for klyron_plugin
use crate::types::PluginConfig;

pub struct PluginBuilder {
    config: PluginConfig,
}

impl PluginBuilder {
    pub fn new() -> Self {
        Self { config: PluginConfig::default() }
    }
    pub fn with_version(mut self, version: &str) -> Self {
        self.config.version = version.to_string();
        self
    }
    pub fn build(self) -> PluginConfig {
        self.config
    }
}
